### PR TITLE
add esil_weak_eq and make $z an operation

### PR DESCRIPTION
**Pull request by condret**
this code is the beginning of the removal of the idea of esil-internal vars as special things doing magic without side-effects. iternal vars become operations which push things on the stack. ppl using #!c and #!lib can override their behaviour using r_anal_esil_set_op. there is going to be no more need to rewrite the parser, if someone wants to create a different kind of analysis on top of esil.

new esil operation ":=" gets introduced, which sets the destination, without touching esil->cur and esil->old. This also enables deprectation of $[0...9].

there are going to be more changes, travis will be red, bc of strcmp like tests on generated esil. worry not

### CODE REVIEW

File: libr/anal/esil.c [hunk 1]
```diff
@@ -495,13 +495,13 @@ static int esil_internal_read(RAnalEsil *esil, const char *str, ut64 *num) {
...
 		break;
-	case 'z': //zero-flag
+/*	case 'z': //zero-flag
 		{
 			ut64 m = genmask (esil->lastsz - 1);
 			*num = (((ut64) esil->cur & m) == 0);
 		}
 		break;
-	case 'b': //borrow
+*/	case 'b': //borrow
```
Comment (radare) [inline libr/anal/esil.c]: dont start lines with */ iif there's a case in there,

**Comment by radare** [comment]
what will happen with $b and all the others $ variables? also $ will be something then?

File: libr/anal/esil.c (hunk 1 above)
Comment (condret) [reply libr/anal/esil.c]: this is a tmp thing, I will remove it later completly

**Comment by condret** [comment]
$c3 => 3,$c

**Comment by condret** [comment]
one cool thing about this is that the "internal vars" now get a "spawnpoint" during execution. NUM is no longer needed, most likely

**Comment by radare** [comment]
lgtm

**Comment by radare** [comment]
pls update the tests

Screenshot 2019-03-29 at 16 19 53

**Comment by radare** [comment]
but ill probably delay this merge after monday/(release)

**Comment by condret** [comment]
let me finish the whole thing, then I look for the tests. Yes, no merge befor release

**Comment by radare** [comment]
Yep. Its marked for 3.5 wont merge yet

File: libr/anal/p/anal_sh.c [hunk 2]
```diff
@@ -855,10 +855,10 @@ static int first_nibble_is_8(RAnal* anal, RAnalOp* op, ut16 code) {
 		if (IS_BT (code)) {
 			r_strbuf_setf (&op->esil, "sr,1,&,?{,0x%x,pc,=,}", op->jump);
 		} else if (IS_BTS (code)) {
-			r_strbuf_setf (&op->esil, "1,$ds,=,sr,1,&,?{,0x%x,pc,=,}", op->jump);
+			r_strbuf_setf (&op->esil, "1,SETD,sr,1,&,?{,0x%x,pc,=,}", op->jump);
 			op->delay = 1; //Only /S versions have a delay slot
 		} else if (IS_BFS (code)) {
-			r_strbuf_setf (&op->esil, "1,$ds,=,sr,1,&,!,?{,0x%x,pc,=,}",op->jump);
+			r_strbuf_setf (&op->esil, "1,SETD,sr,1,&,!,?{,0x%x,pc,=,}",op->jump);
```
Comment (radare) [inline libr/anal/p/anal_sh.c]: SETD?

**Comment by radare** [comment]
pls update the tests and finish the pr asap, dont wanna make those changes in esil take forever

**Comment by radare** [comment]
there are cornflakes^Wconflicts in anal_arm_cs.c pls `git rebase master`

File: libr/anal/p/anal_sh.c (hunk 2 above)
Comment (condret) [reply libr/anal/p/anal_sh.c]: yes, set-delay, internal vars are as vars readonly.

**Comment by radare** [comment]
Fix the remaining test and ill merge

**Pull request merged**